src/main/java/randoop/util/EquivalenceChecker.java

```diff
@@ -0,0 +1,50 @@
+package randoop.util;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Utility class for checking equivalence between objects in randoop.generation.Detective. Mainly
```

Comment:
"Mainly" indicates there is something else.  What else does this do?

Reply:
It performs class equivalence checks (class?> c1,c2; c1.equals(c2)), but the reason for including this class was to handle equivalence checks between primitive types and their boxed counterparts and treat them as equivalent. The comments should be clearer now.
/**
 * Utility class for checking equivalence between objects in randoop.generation.Detective.
 * This class handles the equivalence checking box types and primitive types in addition to
 * the default class equivalence checking.
 */